base/range.jl

```diff
@@ -799,7 +806,7 @@ let smallints = (Int === Int64 ?
     function length(r::OrdinalRange{<:smallints})
         s = step(r)
         isempty(r) && return 0
-        return div(Int(last(r)) - Int(first(r)), s) + 1
+        return Int(div(Int(last(r)) - Int(first(r)), s)) + 1
```

Comment:
This is for `length(::StepRange{Int8,Int128})`. Without this, we'll return a `Int128` for non-empty `r`.
I believe it safe as we convert the `div` result rather than step `s`. `Int` should be enough to represent the `length`